app/controllers/users/authorization_confirmation_controller.rb

```diff
@@ -13,7 +13,11 @@ class AuthorizationConfirmationController < ApplicationController
     def new
       analytics.authentication_confirmation
       @sp = ServiceProvider.find_by(issuer: sp_session[:issuer])
-      @email = EmailContext.new(current_user).last_sign_in_email_address.email
+      @email = if current_user.active_identity_for(@sp)
+                 current_user.active_identity_for(@sp).email_address_for_sharing.email
+               else
+                 current_user.last_sign_in_email_address.email
+               end
```

Comment:
Looks like `active_identity_for(@sp)` will make a DB query, so we might be able to use the nil-safe navigator on this an avoid duplicating the query:

```suggestion
      @email = current_user.active_identity_for(@sp)&.email_address_for_sharing&.email ||
                 current_user.last_sign_in_email_address.email
```